src/auth/strategies/jwt.strategy.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+
+import { UserRepository } from '@src/user/repositories/user.repository';
+import { JwtPayload } from '../types/jwt.interface';
+
+enum MESSAGES {
+  INVALID_TOKEN = 'Invalid Token',
+}
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly userRepository: UserRepository,
+  ) {
+    super({
+      secretOrKey: configService.get<string>('JWT_SECRET'),
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { id } = payload;
+    const user = await this.userRepository.findOne({ where: { id } });
```

Comment:
Here you should call the userService, and the service is the one that has access to the user repo

Reply:
done